File: server/mc/common/src/main/java/calebxzhou/rdi/mc/common/RDI.java
package calebxzhou.rdi.mc.common;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

/**
 * calebxzhou @ 2026-01-06 23:56
 */
public class RDI {
    private static final Logger lgr = LogManager.getLogger("rdi");
    public static final String IHQ_URL;
    public static final String HOST_ID;
    public static final Boolean ALL_OP;
    static {
        String ihqUrl = System.getProperty("rdi.ihq.url");
        if (ihqUrl == null) {
            ihqUrl = "host.docker.internal:65231";
        }
        IHQ_URL = ihqUrl;

        String hostId = System.getenv("HOST_ID");
        if (hostId == null) {
            hostId = System.getProperty("rdi.host.id");
        }
        if (hostId == null) {
            throw new IllegalArgumentException("No HOST_ID provided – stopping");
        }
        HOST_ID = hostId;

        String allOpEnv = System.getenv("ALL_OP");
        if (allOpEnv == null) {
            ALL_OP = false;
        } else {
            ALL_OP = Boolean.parseBoolean(allOpEnv);
        }
    }

}

File: client-mc/1.21.1_neoforge/src/main/kotlin/calebxzhou/rdi/RDI.kt
package calebxzhou.rdi

import net.neoforged.fml.common.Mod
import org.slf4j.LoggerFactory
import org.slf4j.MarkerFactory

val lgr = LoggerFactory.getLogger("rdi")
val logMarker
    get() = { marker: String -> MarkerFactory.getMarker(marker) }

@Mod("rdi")
class RDI {

    companion object {
        @JvmField
        val IHQ_URL =
            System.getProperty("rdi.ihq.url") ?: throw IllegalArgumentException("启动方式错误：找不到服务器地址1")

        @JvmField
        val GAME_IP =
            System.getProperty("rdi.game.ip") ?: throw IllegalArgumentException("启动方式错误：找不到服务器地址2")
        @JvmField
        val HOST_NAME =
            System.getProperty("rdi.host.name") ?: throw IllegalArgumentException("启动方式错误：找不到主机名")

        @JvmField
        val HOST_PORT = System.getProperty("rdi.host.port")?.toInt() ?: throw IllegalArgumentException("启动方式错误：找不到服务器端口")

        @JvmStatic
        fun connectServer(){
            //todo
        }
    }


    init {
        lgr.info("RDI启动中")

    }
}